backend/compact-connect/common_constructs/queued_lambda_processor.py

```diff
@@ -51,14 +51,45 @@ def __init__(
             dead_letter_queue=DeadLetterQueue(max_receive_count=max_receive_count, queue=self.dlq),
         )
 
-        process_function.add_event_source(
-            SqsEventSource(
-                self.queue,
-                batch_size=batch_size,
-                max_batching_window=max_batching_window,
-                report_batch_item_failures=True,
-            ),
+        # The following section of code is equivalent to:
+        # process_function.add_event_source(
+        #     SqsEventSource(
+        #         self.queue,
+        #         batch_size=batch_size,
+        #         max_batching_window=max_batching_window,
+        #         report_batch_item_failures=True,
+        #     ),
+        # )
+        #
+        # Except that we are granting the lambda permission to consume SQS messages via resource policy
+        # on the queue, rather than the more conventional approach of principal policy on the IAM role.
+        #
+        # We use a lower-level add_event_source_mapping method here so that we can control how those
+        # permissions are granted. In this case, we need to grant permissions via resource policy on
+        # the Queue rather than principal policy on the role to avoid creating a dependency from the
+        # role on the queue.
```

Comment:
This detail is great in explaining how it is working, but I still don't understand why we need to avoid creating that dependency. Does it create a circular dependency between stacks? Could we make that detail a little more clear so devs understand the consequence of not doing it this way?